prefect_dask/task_runners.py

```diff
@@ -276,18 +276,24 @@ async def submit(
         else:
             dask_key = key
 
-        self._dask_futures[key] = self._client.submit(
-            call.func,
-            key=dask_key,
-            # Dask defaults to treating functions are pure, but we set this here for
-            # explicit expectations. If this task run is submitted to Dask twice, the
-            # result of the first run should be returned. Subsequent runs would return
-            # `Abort` exceptions if they were submitted again.
-            pure=True,
-            **call_kwargs,
+        self._dask_futures[key] = distributed.as_completed(
```

Comment:
I tried to wrap the future with `as_completed` in the `wait` method but when I do, the behavior is similar to the original code (the completed tasks are retried). I think this is because the future references are held before calling `as_completed`.

Reply:
We should definitely add a comment about _why_ we can't hold references to the future without this wrapper.